### PR TITLE
Add IWSSr-SFLA and Relief measure

### PR DESCRIPTION
**Pull request by MadHypnofrog**

### CODE REVIEW

**Review by LastShekel — COMMENTED** [review]
Add theese algorithms into table in index.rst

File: ITMO_FS/ensembles/ranking_based/Mixed.py [hunk 1]
```diff
@@ -1,42 +1,39 @@
...
+        >>> x, y = make_classification(1000, 50, n_informative = 5, n_redundant = 3, n_repeated = 2, shuffle = True)
+        >>> mixed = Mixed([spearman_corr, pearson_corr])
+        >>> mixed.fit(x, y)
+        >>> mixed.transform(x, 20).shape
+        (1000, 20)
     """
 
-    _filters = []
+    def __init__(self, filters, k, fusion_function=best_goes_first_fusion):
+        self.filters = filters
```
[the remark this answers is not in the document]
Comment (MadHypnofrog) [reply ITMO_FS/ensembles/ranking_based/Mixed.py:29]: check_estimator fails if any parameter name is different from what was passed to init

**Review by LastShekel — CHANGES_REQUESTED** [review]
Also i suggest to add get params and set params

File: ITMO_FS/utils/base_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from sklearn.utils import check_X_y, check_array
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+import pandas as pd
+
+from . import BaseTransformer
+
+class BaseWrapper(BaseTransformer):
```
[the remark this answers is not in the document]
Comment (MadHypnofrog) [reply ITMO_FS/utils/base_wrapper.py:8]: wrappers can transform datasets as well based on selected features, the only thing they add is a predict function

File: ITMO_FS/filters/unsupervised/trace_ratio_laplacian.py [hunk 3]
```diff
@@ -93,12 +90,14 @@ def run(self, X, y):
         e = np.absolute(np.diag(E))
         b = np.absolute(np.diag(B))
         b[b == 0] = 1e-14
-        features_indices = np.argsort(np.divide(b, e))[::-1][0:self.n_selected_features]
-        lam = np.sum(b[features_indices]) / np.sum(e[features_indices])
+        self.selected_features_ = np.argsort(np.divide(b, e))[::-1][0:self.n_features]
+        lam = np.sum(b[self.selected_features_]) / np.sum(e[self.selected_features_])
         prev_lam = 0
         while (lam - prev_lam >= 1e-3):
```
Comment (LastShekel) [inline ITMO_FS/filters/unsupervised/trace_ratio_laplacian.py:99]: Replace that magic number with input value and add todo optimize

File: ITMO_FS/hybrid/IWSSr_SFLA.py [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+        self.relief_iterations = relief_iterations
+        self.sfla_m = sfla_m
+        self.sfla_n = sfla_n
+        self.sfla_q = sfla_q
+        self.s = s
+        self.iterations = iterations
+        self.iterations_leaps = iterations_leaps
+        self.seed = seed
+
+    def generate_frogs(self, weights):
```
Comment (LastShekel) [inline ITMO_FS/hybrid/IWSSr_SFLA.py:79]: I think these functions should be shadowed

File: ITMO_FS/hybrid/IWSSr_SFLA.py [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+                Fitness values for all frogs in the memplex.
+
+            Returns
+            -------
+            array-like, shape (sfla_q) : indices of frogs in the submemplex
+        """
+        order = np.argsort(fitness_values)[::-1]
+        weights = np.zeros(self.sfla_n)
+        for i in range(self.sfla_n):
+            weights[order[i]] = 2 * (self.sfla_n - i) / (self.sfla_n * (self.sfla_n + 1))
```
Comment (LastShekel) [inline ITMO_FS/hybrid/IWSSr_SFLA.py:196]: This could be simplified to (2-2i)/self.sfla_n

**Review by LastShekel — CHANGES_REQUESTED** [review]

File: ITMO_FS/filters/unsupervised/trace_ratio_laplacian.py (hunk 3 above)
Comment (MadHypnofrog) [reply ITMO_FS/filters/unsupervised/trace_ratio_laplacian.py:99]: added

File: ITMO_FS/hybrid/IWSSr_SFLA.py (hunk 4 above)
Comment (MadHypnofrog) [reply ITMO_FS/hybrid/IWSSr_SFLA.py:79]: true, fixed

File: ITMO_FS/hybrid/IWSSr_SFLA.py (hunk 5 above)
Comment (MadHypnofrog) [reply ITMO_FS/hybrid/IWSSr_SFLA.py:196]: (2 - 2i) is a negative number for i > 1 and the existing formula always yields positive values, there's a mistake in your math I guess

**Review by LastShekel — APPROVED** [review]
We merge it

**Pull request merged**